src/libraries/System.Diagnostics.Process/tests/ProcessModuleTests.cs

```diff
@@ -28,6 +30,61 @@ public void TestModuleProperties()
             }
         }
 
+        [ConditionalFact(nameof(IsProcessElevated))]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void TestModuleLongPath()
+        {
+            // // Metadata version: v4.0.30319
+            // .assembly Test
+            // {
+            //   .ver 0:0:0:0
+            // }
+            // .module Test.dll
+            // // MVID: {48E60D10-353B-45DC-AA09-3A1E6B0FD382}
+            // .imagebase 0x00400000
+            // .file alignment 0x00000200
+            // .stackreserve 0x00100000
+            // .subsystem 0x0003       // WINDOWS_CUI
+            // .corflags 0x00000001    //  ILONLY
+            byte[] assemblyImage = Convert.FromBase64String(
```

Comment:
@steveharter @GrabYourPitchforks (as you are the owners of `System.Reflection.Emit`) I remember that Full .NET Framework offered a way to dynamically generate, save to disk, and load from disk a managed assembly. Is this still supported?

We need to test the long process module paths and our current best idea is to dynamically load such an assembly to the test runner process.

Reply:
> save to disk [...] Is this still supported?

No.
https://github.com/dotnet/runtime/issues/15704

Reply:
@stephentoub do you know if we have any existing test that we could just extend? Like something that validates that a .NET executable with path > MAX_PATH can just run for example?

I was thinking about creating a new standalone `.exe` with just a very long name and asserting the module inside. Something like the standalone coreclr regression tests: https://github.com/dotnet/runtime/tree/master/src/tests/Regressions/coreclr

My other idea is to... add a new project with a very long output dll name to the `System.Diagnostics.sln`, reference it from this tests project, and load it dynamically.

I am not sure if this would work if we run our test on machines where long path support is disabled on purpose. In such a case we would have to give it a short name, and after detecting that it's fine (conditional fact), rename the file and then load it. Any thoughts on that?

Reply:
Most or all of our developers have long paths enabled (since it is/was documented as a requirement to build the repo) so I think if the test simply checks for it being enabled that makes sense to me and it would still run fairly often.

Reply:
I've sent a proposal #46685